source/WanManager/wanmgr_interface_sm.c

```diff
@@ -1897,10 +1897,17 @@ static eWanState_t wan_transition_physical_interface_down(WanMgr_IfaceSM_Control
 
     Update_Interface_Status();
 
-    if(p_VirtIf->Reset == TRUE || p_VirtIf->VLAN.Reset == TRUE || p_VirtIf->VLAN.Expired == TRUE)
+    if(pWanIfaceCtrl->WanEnable != FALSE &&
+       pInterface->Selection.Enable != FALSE &&
+       p_VirtIf->Enable != FALSE &&
+       pInterface->Selection.Status != WAN_IFACE_NOT_SELECTED &&
+       pInterface->BaseInterfaceStatus ==  WAN_IFACE_PHY_STATUS_UP)
     {
-        CcspTraceInfo(("%s %d - Interface '%s' - TRANSITION REFRESHING WAN\n", __FUNCTION__, __LINE__, pInterface->Name));
-        return WAN_STATE_REFRESHING_WAN;
+        if(p_VirtIf->Reset == TRUE || p_VirtIf->VLAN.Reset == TRUE || p_VirtIf->VLAN.Expired == TRUE)
```

Comment:
Please reset the flags in wan_state_refreshing_wan itself. Similar to          
    if (pWanIfaceCtrl->WanEnable == FALSE ||
        pInterface->Selection.Enable == FALSE ||
        pInterface->Selection.Status == WAN_IFACE_NOT_SELECTED ||
        p_VirtIf->Enable == FALSE ||
        pInterface->BaseInterfaceStatus !=  WAN_IFACE_PHY_STATUS_UP)
    {
         p_VirtIf->VLAN.Expired = FALSE; //Reset VLAN.Expired
        return wan_transition_physical_interface_down(pWanIfaceCtrl);
    }